controllers/front/availableoffertypes.php

```diff
@@ -24,52 +24,44 @@
  * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
  */
 
-use Comfino\Api;
 use Comfino\ErrorLogger;
+use Comfino\Extended\Api\Serializer\Json as JsonSerializer;
+use Comfino\FinancialProduct\ProductTypesListTypeEnum;
+use Comfino\OrderManager;
+use Comfino\SettingsManager;
 
 if (!defined('_PS_VERSION_')) {
     exit;
 }
 
-require_once _PS_MODULE_DIR_ . 'comfino/src/Api.php';
-require_once _PS_MODULE_DIR_ . 'comfino/src/ErrorLogger.php';
-
 class ComfinoAvailableOfferTypesModuleFrontController extends ModuleFrontController
 {
     public function postProcess()
     {
-        Api::init($this->module);
-        ErrorLogger::init();
+        ErrorLogger::init($this->module);
 
         parent::postProcess();
 
-        $config_manager = new \Comfino\ConfigManager($this->module);
-        $available_product_types = array_map(
-            static function (array $offer_type) { return $offer_type['key']; },
-            $config_manager->getOfferTypes()
-        );
+        header('Content-Type: application/json');
+
+        $serializer = new JsonSerializer();
+        $availableProductTypes = SettingsManager::getProductTypesStrings(ProductTypesListTypeEnum::LIST_TYPE_WIDGET);
 
         if (!Tools::getIsset('product_id')) {
-            echo json_encode($available_product_types);
+            echo $serializer->serialize($availableProductTypes);
             exit;
```

Comment:
Przez `exit` steruje się controllerem? 🤔

Reply:
Tak, też się dziwiłem gdy zobaczyłem to pierwszy raz we wtyczce, którą dostałem do utrzymania. Wywaliłem i szybko zrozumiałem, dlaczego to było użyte. :-D Potem jeszcze z ciekawości zagłębiłem się w kod bazowego kontrolera frontu w PS i wszystko stało się jasne. Dla mnie to taki "ugly hack", ale działa więc i ja to dalej stosuję bez zmian.